src/malli/experimental.cljc

```diff
@@ -47,10 +51,13 @@
     `(let [defn# (c/defn
                    ~name
                    ~@(some-> doc vector)
-                   ~(assoc meta :raw-arglists (list 'quote raw-arglists), :schema schema)
+                   ~(assoc body-meta :raw-arglists (list 'quote raw-arglists), :schema schema)
                    ~@(map (fn [{:keys [arglist prepost body]}] `(~arglist ~prepost ~@body)) parglists)
                    ~@(when-not single (some->> arities val :meta vector)))]
        (m/=> ~name ~schema)
+       ~(when (or (:malli/always-check var-meta)
+                  (:malli/always-check body-meta))
+          `(alter-var-root (var ~name) (fn [f#] (m/-instrument {:schema ~schema} f#))))
```

Comment:
what if we wrapped the body with `m/-instrument` instead?

Reply:
I thought about that but generating the code for the `fn` to put inside `m/-instrument` seemed like a lot of work.

I also considered giving a name to this alter-var-root spell, something like `instrument-var!`

Reply:
I ended up doing this thanks to @dvingo